invokeai/app/util/profiler.py

```diff
@@ -0,0 +1,76 @@
+import cProfile
+from contextlib import suppress
+from logging import Logger
+from pathlib import Path
+from typing import Optional
+
+
+class Profiler:
+    """
+    Simple wrapper around cProfile.
+
+    Usage
+    ```
+      # Create a profiler
+      profiler = Profiler(logger, output_dir)
+      # Start a new profile
+      profiler.new("my_profile")
+      profiler.enable()
+      # Do stuff
+      profiler.disable()
+      profiler.dump()
```

Comment:
Why not just have two methods? I.e. `profiler.start("my_profile")` and `profiler.stop()`.

Reply:
I was trying to retain the flexibility and API of `cProfile.Profile`, but it's definitely less ergonomic.

Also, in the structured stats PR, I added an arg to this dump method to optionally write the stats to JSON. The intention was to output all the profiling data at once, but it couples the stats service to this profiler, which is not great. The two classes shouldn't interact like that.

I agree that just a start and stop on the profiler is better. Can add a method to the stats service to write its data to disk, and just call that at the same time as we write the profiler's data.